lib/jekyll/site_data/directory.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SiteData
+    class Directory
+      # Delegate given (zero-arity) method(s) to the Hash object stored in instance
+      # variable `@meta`.
+      def self.delegate_to_meta(*symbols)
```

Comment:
We do a lot of delegation throughout the Jekyll codebase. Is the ["Forwardable" module](https://ruby-doc.org/stdlib-3.1.2/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegators) insufficient for this?

Reply:
If I remember correctly, the main motivation behind this was to avoid initialising interim objects. At the time, using `Forwardable`'s `def_delegators` method resulted in allocating an `Array` object on every invocation of the delegated method. Anyways, this method has been designated as a `private_class_method` so that it can be changed or removed with minimal side-effects.